File: java/netty_lecture/src/main/java/com/sakura/netty/thrift/PersonServiceImpl.java
package com.sakura.netty.thrift;

import org.apache.thrift.TException;
import thrift.generated.DataException;
import thrift.generated.Person;
import thrift.generated.PersonService;

/**
 * @ClassName : PersonService
 * @Description : thrift生成的接口文件的实现类
 * @Author : Mr.Sakura
 * @Date: 2020-03-19 16:27
 */
public class PersonServiceImpl implements PersonService.Iface {

    @Override
    public Person getPersonByUsername(String username) throws DataException, TException {
        System.out.println("Got Client Param：" + username);

        Person person = new Person().setUsername(username)
                .setAge(18).setMarried(false);
        return person;
    }

    @Override
    public void savePersion(Person person) throws DataException, TException {
        System.out.println("Got Client Param：");
        System.out.println(person.getUsername());
        System.out.println(person.getAge());
        System.out.println(person.isMarried());
    }
}

File: python/thrift-python/PersonServiceImpl.py
# -*- coding:utf-8 -*-

from py.thrift.generated import *

class PersonServiceImpl:
    def getPersonByUsername(self,username):
        print "Python Got client param：" + username

        person = ttypes.Person()
        person.username = username
        person.age = 18
        person.married = False

        return person
    
    def savePersion(self,person):
        print "Python Got client param："

        print person.username
        print person.age
        print person.married